atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/client/creation/DefaultCqlClientFactory.java

```diff
@@ -137,12 +137,15 @@ public DefaultCqlClientFactory() {
             Cluster.Builder builder, CassandraKeyValueServiceConfig config, Set<InetSocketAddress> servers) {
         // Refuse to talk to nodes twice as (latency-wise) slow as the best one, over a timescale of 100ms,
         // and every 10s try to re-evaluate ignored nodes performance by giving them queries again.
-        // Note we are being purposely datacenter-irreverent here, instead relying on latency alone
-        // to approximate what DCAwareRR would do;
-        // this is because DCs for Atlas are always quite latency-close and should be used this way,
-        // not as if we have some cross-country backup DC.
-        LoadBalancingPolicy policy =
-                LatencyAwarePolicy.builder(new RoundRobinPolicy()).build();
+        //
+        // The DCAware RR policy prevents auto-discovery of remote datacenter contact points in a multi-DC setup,
+        // which we are using to orchestrate migrations across datacenters.  We don't expect the policy to be helpful
+        // for latency improvements.
+        //
+        // Since the local DC isn't specified, it will get set based on the contact points we provide.
+        LoadBalancingPolicy policy = LatencyAwarePolicy.builder(
+                        DCAwareRoundRobinPolicy.builder().build())
```

Comment:
I don't think this will work. The DCAwareRoundRobinPolicy needs to know the _name_ of the local datacentre in order to work properly, which isn't accessible to us here.

Reply:
The DCRR policy will auto-discover the datacenter based on the first host to which it connects: https://github.com/datastax/java-driver/blob/3.7.2/driver-core/src/main/java/com/datastax/driver/core/policies/DCAwareRoundRobinPolicy.java#L111

Reply:
Given the autodiscovery code I think this is correct (though admittedly have less Cassandra experience than either of you!)